Changed to "is compiled" rather than "maps" to the DVM op code to avoid ambiguity

jamesray1 opened the pull request:
Maps will be explained later `[here](http://solidity.readthedocs.io/en/develop/types.html#mappings)`_

jamesray1 committed:
Create types.rst

Maps will be explained later `[here](http://solidity.readthedocs.io/en/develop/types.html#mappings)`_

pirapira commented on docs/types.rst:
```diff
@@ -54,7 +54,7 @@ Operators:
 * Arithmetic operators: ``+``, ``-``, unary ``-``, unary ``+``, ``*``, ``/``, ``%`` (remainder), ``**`` (exponentiation), ``<<`` (left shift), ``>>`` (right shift)
 
-Division always truncates (it just maps to the DIV opcode of the EVM), but it does not truncate if both
+Division always truncates (it just maps to the DIV opcode of the EVM. Maps will be explained later `[here](http://solidity.readthedocs.io/en/develop/types.html#mappings)`_), but it does not truncate if both
```
The markup looks like a combination of Markdown and reStructuredText.  This document is written in reStructuredText.  Also, the URL should not appear here.  An older version of the document should not link to the newest version of the document like this.  Use `:ref` role to create a link.

jamesray1 replied:
Yeah sorry I've since figured out that it's meant to be of this format:

`[here](http://solidity.readthedocs.io/en/develop/types.html#mappings)`_

chriseth commented:
The term "maps" in this sentence is completely unrelated to the term "mapping", it rather means "is compiled to".

jamesray1 commented:
@chriseth OK thanks for the clarification.

jamesray1 committed:
Create types.rst

"is compiled" avoids ambiguity compared with "maps", especially for those who may not understand what maps may mean and may confuse it with the "mapping" code keyword

jamesray1 commented:
Please see my updated commit https://github.com/ethereum/solidity/pull/2373/commits/c1e0a6557dd7590f5ce49c9cf742fdc5ddd9ca38 based on @chriseth's clarification above.
